mautic/mautic
# Create company from ip lookup switch

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Enhancement
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

In PR https://github.com/mautic/mautic/pull/7083 I've added option to create and link a company from IP lookup info to Mautic. It may cause problems as if a contact has a company and nothing else then Mautic considers it as identified contact. That may not be the best option for all users. Therefore this PR adds a new option which will allow users to decide what behavior they want.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. This will be a bit hard to test as only some paid IP lookup services provide the company data and only some IP addresses have it. But if you can then test that IP lookup still works. The change is covered with tests.
3. Check that the new option is in the configuration and you can turn it on and off.
